server/test/tests.go

```diff
@@ -0,0 +1,256 @@
+		Specs       []TestSpec                   `json:"specs"`
+		Outputs     maps.Ordered[string, Output] `json:"outputs"`
+		Summary     Summary                      `json:"summary"`
+	}
+
+	Output struct {
+		Selector SpanQuery `json:"selector"`
+		Value    string    `json:"value" expr_enabled:"true"`
+	}
+
+	TestSpec struct {
+		Selector   Selector    `json:"selector"`
+		Name       string      `json:"name,omitempty"`
+		Assertions []Assertion `json:"assertions"`
+	}
+
+	Selector struct {
+		Query          SpanQuery    `json:"query"`
+		ParsedSelector SpanSelector `json:"parsedSelector"`
+	}
+
+	SpanSelector struct {
+		Filters       []SelectorFilter     `json:"filters"`
+		PseudoClass   *SelectorPseudoClass `json:"pseudoClass,omitempty"`
+		ChildSelector *SpanSelector        `json:"childSelector,omitempty"`
+	}
+
+	SelectorFilter struct {
+		Property string `json:"property"`
+		Operator string `json:"operator"`
+		Value    string `json:"value"`
+	}
+
+	SelectorPseudoClass struct {
+		Name     string `json:"name"`
+		Argument *int32 `json:"argument,omitempty"`
+	}
+
+	Summary struct {
+		Runs    int     `json:"runs"`
+		LastRun LastRun `json:"lastRun"`
+	}
+
+	LastRun struct {
+		Time   time.Time `json:"time"`
+		Passes int       `json:"passes"`
+		Fails  int       `json:"fails"`
+	}
+
+	SpanQuery string
+
+	Assertion string
+
+	AssertionExpression struct {
+		LiteralValue LiteralValue
+		Operation    string
+		Expression   *AssertionExpression
+	}
+
+	LiteralValue struct {
+		Value string
+		Type  string
+	}
+
+	RunMetadata map[string]string
+
+	Run struct {
+		ID          int
+		TestID      id.ID
+		TestVersion int
+
+		// Timestamps
+		CreatedAt                 time.Time
+		ServiceTriggeredAt        time.Time
+		ServiceTriggerCompletedAt time.Time
+		ObtainedTraceAt           time.Time
+		CompletedAt               time.Time
+
+		// trigger params
+		State   RunState
+		TraceID trace.TraceID
+		SpanID  trace.SpanID
+
+		// result info
+		TriggerResult trigger.TriggerResult
+		Results       *RunResults
+		Trace         *model.Trace
+		Outputs       maps.Ordered[string, RunOutput]
+		LastError     error
+		Pass          int
+		Fail          int
+
+		Metadata RunMetadata
+
+		// environment
+		Environment environment.Environment
+
+		// transaction
+
+		TransactionID    string
+		TransactionRunID string
+		Linter           model.LinterResult
+	}
+
+	RunResults struct {
+		AllPassed bool
+		Results   maps.Ordered[SpanQuery, []AssertionResult]
+	}
+
+	RunOutput struct {
+		Name     string
+		Value    string
+		SpanID   string
+		Resolved bool
+		Error    error
+	}
+
+	AssertionResult struct {
+		Assertion Assertion
+		AllPassed bool
+		Results   []SpanAssertionResult
+	}
+
+	SpanAssertionResult struct {
+		SpanID        *trace.SpanID
+		ObservedValue string
+		CompareErr    error
+	}
+)
+
+func (t Test) GetID() id.ID {
+	return t.ID
+}
+
+func (t Test) Validate() error {
+	return nil
+}
+
+func (sar SpanAssertionResult) SafeSpanIDString() string {
```

Comment:
Can we rename this? the `safe` is implied by the return type. it's a string, which always has a value, and no error
```suggestion
func (sar SpanAssertionResult) SpanIDString() string {
```